### PR TITLE
feat(database): enable database multi-resource support

### PR DESCRIPTION
**Pull request by jshcrowthe**
This enables database to leverage multi instance feature enabling multiple database instances per app.

### CODE REVIEW

File: src/database/core/RepoManager.ts [hunk 1]
```diff
@@ -121,12 +130,21 @@ export class RepoManager {
...
+    if (!appRepos) {
+      appRepos = {};
+      this.repos_[app.name] = appRepos;
+    }
+
+    let repo = safeGet(appRepos, repoInfo.toURLString());
     if (repo) {
-      fatal('FIREBASE INTERNAL ERROR: Database initialized multiple times.');
+      fatal(
+        'Database initialized multiple times. Please make sure the format of the database URL matches with each database() call.'
```
Comment (mikelehen) [inline src/database/core/RepoManager.ts]: Alternatively we could just return the existing instance.  Would that cause other problems though?  I guess app would end up with multiple references to it and might call INTERNAL.delete() multiple times, etc.?

I guess I don't really feel strongly. It's probably not too harmful to make them pass the exact same URL every time.

File: src/database/core/RepoManager.ts [hunk 2]
```diff
@@ -34,10 +34,12 @@ let _staticInstance: RepoManager;
  */
 export class RepoManager {
   /**
-   * @private {!Object.<string, !Repo>}
+   * @private {!Object.<string, Object<string, !fb.core.Repo>>}
    */
   private repos_: {
-    [name: string]: Repo;
+    [name: string]: {
+      [name: string]: Repo;
```
[the remark this answers is not in the document]
Comment (jshcrowthe) [reply src/database/core/RepoManager.ts]: Sure, no problem.

File: src/database/core/RepoManager.ts [hunk 3]
```diff
@@ -55,14 +57,18 @@ export class RepoManager {
 
   // TODO(koss): Remove these functions unless used in tests?
   interrupt() {
-    for (const repo in this.repos_) {
-      this.repos_[repo].interrupt();
+    for (const instance in this.repos_) {
+      for (const repo in this.repos_[instance]) {
```
[the remark this answers is not in the document]
Comment (jshcrowthe) [reply src/database/core/RepoManager.ts]: Yup, changed this and the function below.

File: src/database/core/RepoManager.ts (hunk 1 above)
Comment (jshcrowthe) [reply src/database/core/RepoManager.ts]: Yeah, I don't know that I'm terribly heartbroken making users pass the extra string around, especially considering that they can still just use the project config instead.

**Pull request merged**